Feature/tao 10203 advanced search

mike-ionut-mihai-sandu-tao opened the pull request:
WIP please don't merge !!!!!!!!!!!!!!!!!!!!!!

created pr for checking the code purposes

andreluizmachado committed:
Merge branch 'feature/TAO-10203-advanced-search' of https://github.com/oat-sa/tao-core into fix/TAO-10423-label-not-indexed-in-memory-queue

andreluizmachado committed:
TAO-10423 Added unit tests for default behaviour on hasClassSupport method

andreluizmachado committed:
TAO-10423 Added unit tests for default behaviour on hasClassSupport method

andreluizmachado committed:
Merge pull request #2646 from oat-sa/fix/TAO-10423-label-not-indexed-in-memory-queue

TAO-10423 added EventAggregatorService

adriloma committed:
chore(search-component): updated component to trigger query builder on enter keypress

adriloma committed:
chore(search): updated search bar component in tao action bar to use new search component

adriloma committed:
chore(search): implemented query search retention

adriloma committed:
chore(search): implemented searchStore management

andreluizmachado committed:
TAO-10287 reindexed resources when moving classes

andreluizmachado committed:
TAO-10287 removed debug

andreluizmachado committed:
TAO-10287 added one more test case

andreluizmachado committed:
TAO-10287 added one more test case

andreluizmachado committed:
TAO-10287 added one more test case

adriloma committed:
chore(search-component): full refactor of search component to use new search modal component and search retention

andreluizmachado committed:
Merge branch 'feature/TAO-10287-implement-event-resource-moved' of https://github.com/oat-sa/tao-core into feature/TAO-10287-implement-event-resource-moved

andreluizmachado committed:
TAO-10287 Added namespaces on old unit tests, added IndexIteratorFactory test case

andreluizmachado committed:
TAO-10287 fixed broken imports

andreluizmachado committed:
TAO-10287 fixed broken unit test

andreluizmachado committed:
TAO-10287 fix broken unit test

andreluizmachado committed:
TAO-10287 fix broken unit test

andreluizmachado committed:
TAO-10287 fix CS

andreluizmachado committed:
TAO-10495 added totalCount on response

adriloma committed:
chore(code-quality): updated comments and variable names to more accurate ones

adriloma committed:
Merge branch 'feature/TAO-10495-add-totalItemCount-on-search-results-endpoint' into feature/TA0-10453/implement-query-builder

adriloma committed:
chore(search-retention): display total number of records instead of current page records

adriloma committed:
chore(code-quality): minor changes

andreluizmachado committed:
Merge pull request #2659 from oat-sa/feature/TAO-10495-add-totalItemCount-on-search-results-endpoint

TAO-10495 expose totalCount attribute on search endpoint

andreluizmachado committed:
TAO-10287 Fixed PR issues

andreluizmachado committed:
TAO-10366 - start added event listener

adriloma committed:
chore(search-modal): updated exepcted events to match searchModal namespace

adriloma committed:
chore(styles): added styles on focus

andreluizmachado committed:
TAO-10366 Indexed documents when permission access changed

andreluizmachado committed:
TAO-10366 updated privileges into one query

andreluizmachado committed:
fixed unit tests

andreluizmachado committed:
fixed unit tests

andreluizmachado committed:
Merge pull request #2657 from oat-sa/feature/TAO-10287-implement-event-resource-moved

TAO-10287 reindexed resources when moving classes

adriloma committed:
chore(action-manager): add method to get resourceContext

adriloma committed:
chore(code-improvements): fix bad practises consdering PR comments

adriloma committed:
Merge branch 'feature/TAO-10453/implement-query-builder' into feature/TAO-10457/implement-class-filter

adriloma committed:
chore(search-component): implemented function to get rootClassUri

adriloma committed:
chore(class-filter): included missing contexts to getRootClassUri

adriloma committed:
chore(code-style): improved code style

Commit:
Merge pull request #2660 from oat-sa/feature/TAO-10453/implement-query-builder

Feature/tao 10453/implement query builder

adriloma committed:
Merge branch 'feature/TAO-10453/implement-query-builder' into feature/TAO-10457/implement-class-filter

adriloma committed:
chore(class-filter): implemented filter into searchStore

andreluizmachado committed:
TAO-10366 Added unit tests

andreluizmachado committed:
TAO-10366 refactory of condition on DataAccessControlChangedListener

andreluizmachado committed:
TAO-10366 added unit tests, fixed tests namespace

andreluizmachado committed:
TAO-10366 PHPDoc fix

adriloma committed:
chore(search-component): removed harcoded rootClassUri

adriloma committed:
chore(search-component): removed unnecessary constant

adriloma committed:
chore(search-modal): use urlHelper parser to get rootClassUri

Commit:
Merge pull request #2668 from oat-sa/feature/TAO-10457/implement-class-filter

Feature/tao 10457/implement class filter

adriloma committed:
chore(version): bump tao-core-ui-fe version to get class filter on searchModal component

andreluizmachado committed:
TAO-10203 fixed conflicts, merge with master

adriloma committed:
Merge branch 'feature/TAO-10203-advanced-search' into feature/TAO-10457/implement-class-filter-bump-version

Commit:
Merge pull request #2672 from oat-sa/feature/TAO-10457/implement-class-filter-bump-version

chore(version): bump tao-core-ui-fe version

adriloma committed:
chore(search): removed deprecated search helper and styles updated

Commit:
Merge pull request #2676 from oat-sa/feature/TAO-10453/remove-deprecated-search-tooltip

feat(search): removed deprecated search helper and styles updated

andreluizmachado committed:
TAO-10366 fix:only filter by permissions when result is not empty

andreluizmachado committed:
TAO-10366 fix conflicts, merge with feature/TAO-10203-advanced-search

andreluizmachado committed:
TAO-10366 fix:only filter by permissions when result is not empty

andreluizmachado committed:
TAO-10366 - renamed Listener methods to kepp consistency

andreluizmachado committed:
TAO-10366 Refactory - added getOperations method

andreluizmachado committed:
TAO-10366 Refactory - removed context variables from logs, it is causing warnings on TAOLog class

andreluizmachado committed:
Merge pull request #2684 from oat-sa/feature/TAO-10366-create-event-for-changing-resource-privileges

Feature/tao 10366 create event for changing resource privileges

andreluizmachado committed:
TAO-10366 fix resources on root class were not begin updated

andreluizmachado committed:
TAO-10366 fix resources on root class were not begin updated

andreluizmachado committed:
Merge pull request #2686 from oat-sa/feature/TAO-10366-create-event-for-changing-resource-privileges

Feature/tao 10366 create event for changing resource privileges

adriloma committed:
fix(search): change click event handler

Commit:
TAO-10473 creating structure for new endpoint

Commit:
TAO-10473 Already getting data related to class, needs filtering and better implementation

Commit:
Merge pull request #2687 from oat-sa/fix/TAO-10546/fix-trigger-search-by-click-icon

fix(search): change click event handler

Commit:
TAO-10473 Updated service and registering

Commit:
TAO-10473 Added count method to value collection repositories and service

Commit:
TAO-10473 Added parameter for maxListSize and URL to retrieve property options

Commit:
TAO-10473 Created DO for metadata response

Commit:
TAO-10473 Last adjustments

Commit:
TAO-10473 Adjustments and tests

Commit:
TAO-10473 Added collections for classes and metadata

Commit:
TAO-10473 Added tests for service/collections

Commit:
TAO-10473 Add end blank line to some classes

Commit:
TAO-10473 CR changes

Commit:
Merge pull request #2692 from oat-sa/feature/TAO-10473-metadata-discovery-endpoint

Feature/tao 10473 metadata discovery endpoint

adriloma committed:
chore(bump-version): bump tao-core-ui version from 1.10.1 to 1.10.5

bartlomiejmarszal commented on actions/class.ClassMetadata.php:
```diff
@@ -0,0 +1,44 @@
+use oat\tao\model\Lists\Business\Service\ClassMetadataService;
+use oat\tao\model\Lists\Presentation\Web\RequestHandler\ClassMetadataSearchRequestHandler;
+
+class tao_actions_ClassMetadata extends tao_actions_CommonModule
```
I think we should avoid creating classes without namespaces.

bartlomiejmarszal reviewed (commented):
We are introducing almost 6k new lines of code inside of tao-core. Part of this code is strictly related to Advanced Search functionality.

Commit:
TAO-10473 Adjusted filename

Commit:
TAO-10473 Adjusted check for maxListSize parameter

Commit:
TAO-10473 Adjusted URL template to retrieve property items

boajer commented on actions/class.PropertiesAuthoring.php:
```diff
@@ -461,4 +468,63 @@ protected function extractPropertyData($data)
+    {
+        $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
+
+        return get_class($searchService) === \oat\tao\elasticsearch\ElasticSearch::class;
```
1. New dependency for `core` from some lib can be ( and should be ) avoided here
2. Verifying classname just increasing coupling

gabrielfs7 replied:
This will also break the code in case lib is not installed.

boajer replied:
fix https://github.com/oat-sa/tao-core/pull/2834/files

Commit:
TAO-10473 Adjusted filter for custom properties

Commit:
TAO-10473 CR suggestions

Commit:
TAO-10473 CR suggestions

Commit:
Merge pull request #2699 from oat-sa/feature/TAO-10473-metadata-discovery-endpoint

TAO-10473 Fixes for new metadata discovery endpoint

bartlomiejmarszal commented on migrations/Version202010042003512234_tao.php:
```diff
@@ -0,0 +1,31 @@
+
+    public function up(Schema $schema): void
+    {
+        (new RegisterClassMetadataServices())
```
This migration is responsible only for setting service locator inside new object?

bartlomiejmarszal replied:
We should just register service using ServiceManager instead of using other class that is going to do this for us.

bartlomiejmarszal commented on models/classes/TaoOntology.php:
```diff
@@ -30,6 +30,7 @@ interface TaoOntology
     const CLASS_URI_SUBJECT = 'http://www.tao.lu/Ontologies/TAOSubject.rdf#Subject';
     const CLASS_URI_TEST = 'http://www.tao.lu/Ontologies/TAOTest.rdf#Test';
     const CLASS_URI_DELIVERY = 'http://www.tao.lu/Ontologies/TAODelivery.rdf#Delivery';
+    const CLASS_URI_ASSEMBLED_DELIVERY = 'http://www.tao.lu/Ontologies/TAODelivery.rdf#AssembledDelivery';
```
Why this was added here?

bartlomiejmarszal commented on models/classes/dto/OldProperty.php:
```diff
@@ -0,0 +1,49 @@
+
+use core_kernel_classes_Resource;
+
+class OldProperty
```
This is a misleading name. We are using this when we have to determine if our property (`core_kernel_classes_Property`) has changed

bartlomiejmarszal commented on models/classes/resources/ResourceWatcher.php:
```diff
@@ -114,4 +133,80 @@ public function getUpdatedAt(\core_kernel_classes_Resource $resource)
+        return false;
+    }
+
+    private function hasClassSupport(core_kernel_classes_Resource $resource): bool
```
This is a misleading name. We are checking here if a resource is a class

bartlomiejmarszal commented on models/classes/resources/ResourceWatcher.php:
```diff
@@ -114,4 +133,80 @@ public function getUpdatedAt(\core_kernel_classes_Resource $resource)
+        return $resource instanceof core_kernel_classes_Class;
+    }
+
+    private function ignoreEditIemClassUpdates(): bool
```
What is this class doing?

bartlomiejmarszal commented on models/classes/resources/ResourceWatcher.php:
```diff
@@ -114,4 +133,80 @@ public function getUpdatedAt(\core_kernel_classes_Resource $resource)
+        }
+    }
+
+    private function hasResourceSupport(core_kernel_classes_Resource $resource): bool
```
What is this method responsibility?

bartlomiejmarszal commented on models/classes/search/strategy/GenerisIndexUpdater.php:
```diff
@@ -0,0 +1,48 @@
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\search\index\IndexUpdaterInterface;
+
+class GenerisIndexUpdater extends ConfigurableService implements IndexUpdaterInterface
```
Are we going to implement this? What is a purpose of this class?

bartlomiejmarszal commented on actions/class.Search.php:
```diff
@@ -98,17 +98,19 @@ public function search(Search $searchService, PermissionHelper $permissionHelper
 
         $totalPages = is_null($rows) ? 1 : ceil($results->getTotalCount() / $rows);
 
-        $results = $results->getArrayCopy();
```
Can we revert those changes?

bartlomiejmarszal commented on actions/form/class.Clazz.php:
```diff
@@ -83,22 +93,30 @@ protected function getTopClazz()
+        $options = [
+            'index' => $index,
+            'isParentProperty' => $isParentProp,
+            'disableIndexChanges' => $this->disableIndexChanges
```
This should be configurable option

bartlomiejmarszal commented on config/default/IndexUpdater.conf.php:
```diff
@@ -0,0 +1,24 @@
+
+use oat\tao\model\search\strategy\GenerisIndexUpdater;
+
+return new GenerisIndexUpdater();
```
This is setting default index updater when we don't know if we want to have any Indexer, no?

bartlomiejmarszal commented on models/classes/search/Search.php:
```diff
@@ -58,7 +58,7 @@ public function flush();
      *
      * @param IndexIterator|array $documents
-     * @return boolean true if successfully indexed
+     * @return int count of indexed documents
```
Is this confirmed information?

adriloma committed:
chore(keyboard-navigation): implemented opening search modal by ctrl + k shortcut

adriloma committed:
chore(keyboard-navigation): added namespace to prevent unbinding from other components

adriloma committed:
chore(keyboard-navigation): bind event to document instead of window

bartlomiejmarszal commented on models/classes/search/index/IndexService.php:
```diff
@@ -65,105 +62,52 @@ public function runIndexing()
+     * return IndexDocumentBuilderInterface
      */
-    public function createDocumentFromResource(\core_kernel_classes_Resource $resource)
+    public function getDocumentBuilder(): IndexDocumentBuilderInterface
```
This looks for me like a breaking change. This service does not have interface though but it may be implemented elsewhere.

bartlomiejmarszal replied:
How this influence this part of code:
https://github.com/oat-sa/extension-tao-outcomeui/blob/4aaf92dd713750ddb30d37d9474828a1cc52c0e5/model/search/ResultIndexIterator.php#L252

bartlomiejmarszal commented on test/unit/extension/UpdateExtensionsTest.php:
```diff
@@ -128,7 +128,8 @@ private function getExtensionManagerMock()
         $extensionFoo->method('getManifest')
             ->willReturn($fooManifest);
-
+        $extensionFoo->method('getDir')
```
Why is this change introduced?

adriloma committed:
chore(package-json): bump tao-core-ui fe version

Commit:
Merge pull request #2708 from oat-sa/feature/TAO-10592/search-keyboard-navigation

Feature/tao 10592/search keyboard navigation

shpran committed:
Feature/CON-21/Advanced Search flag mechanism

shpran committed:
Feature/CON-21/Bump version

shpran committed:
Feature/CON-21/Update README.md

shpran committed:
Feature/CON-21/Remove unnecessary PHPDocs

bartlomiejmarszal commented on scripts/tools/index/IndexPopulator.php:
```diff
@@ -0,0 +1,197 @@
+
+        file_put_contents($this->getOption('lock'), $class->getUri() . PHP_EOL . 'FINISHED');
+
+        return $this->getScriptReport($result, $reportedClass, $limit, $offset);
```
This will break a loop

shpran committed:
Feature/CON-21/Check for elastic search configuration. Fix test.

shpran committed:
Feature/CON-21/Change grant access to back_office level

shpran committed:
Feature/CON-21/Fix migration messages and description

shpran committed:
Feature/CON-21/Fix test

bartlomiejmarszal committed:
ClassMetadata is missing acl

gabrielfs7 committed:
Merge pull request #2723 from oat-sa/feature/con-23/missing_acl_rule

ClassMetadata is missing acl

shpran committed:
Feature/CON-21/Use negative feature flag instead of positive

adriloma committed:
chore(package-json): bump tao-core-ui-fe to 1.13.0 (keyboard navigation)

Commit:
merge develop

bartlomiejmarszal committed:
Indexing parent tree string (#2722)

Index item parent classes

shpran committed:
Feature/CON-21/Bump version

bartlomiejmarszal committed:
tao-core-ui version bump

Commit:
Merge pull request #2726 from oat-sa/feature/con-23/tao-core-ui-update

tao-core-ui version bump

mike-ionut-mihai-sandu-tao committed:
Merge pull request #2719 from oat-sa/feature/con-21/advanced-search-flag-mechanism

Feature/CON-21/Advanced Search flag mechanism

Commit:
CON-18 update

Commit:
Merge branch 'feature/TAO-10203-advanced-search' into feature/CON-18/enable-disabled-add-criteria

Commit:
Merge pull request #2728 from oat-sa/feature/CON-18/enable-disabled-add-criteria

[Feature] CON 18 - Enable disabled add criteria

bartlomiejmarszal committed:
searchModal will call universal query using proxy and bridge

bartlomiejmarszal committed:
searchQuery object is now accepting null as `rows` and `page`
SearchQueryFactory is missing in repo
SearchQueryFactoryTest
required changes in search

bartlomiejmarszal committed:
SearchQueryFactoryTest

bartlomiejmarszal committed:
Missing empty line

bartlomiejmarszal committed:
PSR-12

bartlomiejmarszal committed:
Avoid double call
fix code style in GenerisSearchBridgeTest

bartlomiejmarszal committed:
rename createSearchQuery to create

bartlomiejmarszal committed:
SearchQueryTest - PSR-12

bartlomiejmarszal committed:
SearchQueryFactory - PSR-12

bartlomiejmarszal committed:
Frontend update

bartlomiejmarszal committed:
Extracted ResultSetResponseNormalizer

bartlomiejmarszal committed:
Add missing php headers

bartlomiejmarszal committed:
Update test/unit/models/classes/search/SearchProxyTest.php

Co-authored-by: Gabriel Felipe Soares <gabrielfs7@gmail.com>

bartlomiejmarszal committed:
Missing empty line

bartlomiejmarszal committed:
EMpty lines

bartlomiejmarszal committed:
Fix tests, spelling fix

bartlomiejmarszal committed:
Merge branch 'feature/con-134/search-modal-universal-query' of https://github.com/oat-sa/tao-core into feature/con-134/search-modal-universal-query

boajer committed:
Merge pull request #2735 from oat-sa/feature/con-134/search-modal-universal-query

Feature/con 134/search modal universal query

bartlomiejmarszal committed:
Merge branch 'develop' into feature/TAO-10203-advanced-search

bartlomiejmarszal committed:
npm install on develop merge

mike-ionut-mihai-sandu-tao committed:
removed condition and added check if there is a widget

mike-ionut-mihai-sandu-tao committed:
extracted variable

mike-ionut-mihai-sandu-tao committed:
renamed the var

mike-ionut-mihai-sandu-tao committed:
Merge pull request #2739 from oat-sa/fix/CON-176/fixed-criteria-selction

removed condition and added check if there is a widget

bartlomiejmarszal committed:
Check if property has widget first

bziondik committed:
hide resource selector in search modal for results page

bziondik committed:
use @oat-sa/tao-core-ui working branch

bziondik committed:
save delivery result id

bziondik committed:
check isResultPage from context

bziondik committed:
update @oat-sa/tao-core-ui

bartlomiejmarszal committed:
Merge branch 'feature/CON-76/advance-search-in-result-page' of https://github.com/oat-sa/tao-core into feature/CON-76/advance-search-in-result-page

bziondik committed:
update @oat-sa/tao-core-ui

bartlomiejmarszal committed:
searchParams setSuccessJsonResponse

bartlomiejmarszal committed:
Merge branch 'feature/CON-76/advance-search-in-result-page' of https://github.com/oat-sa/tao-core into feature/CON-76/advance-search-in-result-page

bziondik committed:
update @oat-sa/tao-core-ui

bartlomiejmarszal committed:
ResultSetMapper created
Normalize takes new param structure

bartlomiejmarszal committed:
Guard if request has structure, query, rootNode, parentNode
try if can build model promise

bartlomiejmarszal committed:
Remove noinspection annotation
refactor ClassMetadataServiceTest

bartlomiejmarszal committed:
Test if resource has a widget

bartlomiejmarszal committed:
Migration for ResultSetMapper

bartlomiejmarszal committed:
ResultSetMapperTest added

bartlomiejmarszal committed:
Change key for service

bartlomiejmarszal committed:
uri is not key anymore. I will create object with id as a attribute

bziondik committed:
send placeholder to searchModel, store delivery result id

bartlomiejmarszal committed:
uri is not key anymore. I will create object with id as a attribute

bartlomiejmarszal committed:
Merge branch 'feature/CON-76/advance-search-in-result-page' of https://github.com/oat-sa/tao-core into feature/CON-76/advance-search-in-result-page

bartlomiejmarszal committed:
Remove non existing uses

bziondik committed:
update "@oat-sa/tao-core-ui": "1.19.0"

bartlomiejmarszal committed:
Update default config file for ResultSetMapper

bartlomiejmarszal committed:
Merge branch 'feature/CON-76/advance-search-in-result-page' of https://github.com/oat-sa/tao-core into feature/CON-76/advance-search-in-result-page

bartlomiejmarszal committed:
Update default config file for ResultSetMapper

bartlomiejmarszal committed:
response['data'][]

bartlomiejmarszal committed:
ResultSetMapper - code refactor

bartlomiejmarszal committed:
ResultSetMapper - reorder default config

bartlomiejmarszal committed:
GenerisSearch returning results in correct format

bartlomiejmarszal committed:
ResultSetMapper change method names corresponding to class name

bartlomiejmarszal committed:
Removed mapResultSetModel from ResultSetMapper
create ResultSetFilter

bziondik committed:
id on Result page in case GenerisSearch, ElasticSearch

bartlomiejmarszal committed:
Use ElasticSearch checker in order to provide more advance promise model for resultSet.

bartlomiejmarszal committed:
Merge branch 'feature/CON-76/advance-search-in-result-page' of https://github.com/oat-sa/tao-core into feature/CON-76/advance-search-in-result-page

bartlomiejmarszal committed:
migration update

bartlomiejmarszal committed:
strict comparison in `in_array`

bartlomiejmarszal committed:
In elastic search type should be a structure variable from request

bartlomiejmarszal committed:
optionStructureMap structure change

bartlomiejmarszal committed:
Fix ResultSetMapperTest

bartlomiejmarszal committed:
ResultSetResponseNormalizer will not show resources that user has no permission to access, fix test

bartlomiejmarszal committed:
return type in getResultSetData

bartlomiejmarszal committed:
Rename test method name to correspond tested method name

bartlomiejmarszal committed:
migration update

bartlomiejmarszal committed:
missing extra line from PSR-12

mike-ionut-mihai-sandu-tao committed:
Merge pull request #2743 from oat-sa/feature/CON-76/advance-search-in-result-page

Feature/CON-76/advance search in result page

bartlomiejmarszal committed:
npm update

Commit:
Merge pull request #2750 from oat-sa/fix/con-76/fix-fe-button-name

npm update

Commit:
Merge develop into feature/TAO-10203-advanced-search

bartlomiejmarszal committed:
Merge pull request #2766 from oat-sa/Merge/develop-into-TAO-10203-advanced-search

Merge develop into TAO-10203-advanced-search

Commit:
fix: CON-122 - search field buttons focus style

Commit:
build frontend

Commit:
update dependencies

Commit:
bump version

Commit:
Revert "update dependencies"

This reverts commit d5e40b8317ae7092e0db77c7de01c7fcecfa40e0.

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch fix/CON-122/graphic-design-match-mockup
# Your branch is up to date with 'origin/fix/CON-122/graphic-design-match-mockup'.
#
# Changes to be committed:
#	modified:   views/package-lock.json
#	modified:   views/package.json
#

Commit:
Revert "bump version"

This reverts commit 566c66a8b399a1fb510c7b76ed61497304b4f2d4.

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch fix/CON-122/graphic-design-match-mockup
# Your branch is up to date with 'origin/fix/CON-122/graphic-design-match-mockup'.
#
# Changes to be committed:
#	modified:   manifest.php
#

Commit:
update dependencies

tereshhhchenko committed:
Merge pull request #2768 from oat-sa/fix/CON-122/graphic-design-match-mockup

[Fix] CON 122 - Graphic design match mockup

tereshhhchenko committed:
[Fix] CON-373 advanced search UI adjustments (#2769)

* fix: CON-373 search icons alignment and results counter font

Commit:
chore: Bump manifest file

Commit:
chore: Update package version with non-merged changes

mike-ionut-mihai-sandu-tao committed:
build: merged develop

Commit:
chore: update package version

Commit:
Merge branch 'feature/TAO-10203-advanced-search' into feature/CON-336/search-by-criteria-without-text

Commit:
Merge pull request #2779 from oat-sa/feature/CON-336/search-by-criteria-without-text

Feature [CON-336]: Be able to search by criteria alone, without a search text

Commit:
Bump manifest version

bartlomiejmarszal committed:
Add events

bartlomiejmarszal committed:
missing end of line

tereshhhchenko committed:
[Fix] CON-122 - advanced search modal match mockup  (#2772)

* fix: CON-122 graphic-design-match-mockup

gabrielfs7 committed:
feat: Extend migration tasks configs

* refact: allow max id to be overriden

* fix: fix wrong chunk calculation

* chore: add unit tests

* chore: add unit tests

* chore: add deprecation notice

* chore: update date and comments

* chore: update comments

Co-authored-by: Gabriel Felipe Soares <gabriel.felipe.soares@taotesting.com>

bartlomiejmarszal committed:
Moved event trigger from generis to addClassProperty

bartlomiejmarszal committed:
Moved event trigger from generis to addClassProperty

bartlomiejmarszal committed:
Merge remote-tracking branch 'origin/feature/CON-495/property-action-events' into feature/CON-495/property-action-events

# Conflicts:
#	actions/class.PropertiesAuthoring.php

bartlomiejmarszal committed:
ClassPropertyDeletedEvent now takes class in constructor
Missing tests

bartlomiejmarszal committed:
ClassPropertyDeletedEvent take array as second argument

bartlomiejmarszal committed:
Create AbstractClassPropertyEvent
add class to properties when property changes

bartlomiejmarszal committed:
fix abstract class

Commit:
chore: bump dependency version

bartlomiejmarszal committed:
Merge branch 'feature/TAO-10203-advanced-search' into feature/CON-495/property-action-events

bartlomiejmarszal committed:
Move events to generis

bartlomiejmarszal committed:
Move trigger to generis

bartlomiejmarszal committed:
Move Class property tests

bartlomiejmarszal committed:
Fix formatting

Commit:
fix: advanced search bool criteria labels positioning and semantics

tereshhhchenko committed:
Merge pull request #2788 from oat-sa/fix/CON-534/advanced-search-criteria-labels

[Fix] CON-534 Advanced search bool criteria labels positioning and semantics

bartlomiejmarszal committed:
Fix class property change event

Commit:
refact: use same strategy to get metadata

Commit:
refact: use same strategy to get metadata

bartlomiejmarszal committed:
Use classMetadataValuesService to get properties

bartlomiejmarszal committed:
Use classMetadataValuesService to get properties

bartlomiejmarszal committed:
getByClassExplicitly class that will be search not recursively, used by MetadataNormalizer

bartlomiejmarszal committed:
GetClassMetadataValuesServiceTest created

bartlomiejmarszal committed:
Radiobox test

bartlomiejmarszal committed:
getByClass => getByClassRecursive

bartlomiejmarszal committed:
getByClassExplicitly is not recursive

bartlomiejmarszal committed:
Merge pull request #2783 from oat-sa/feature/CON-495/property-action-events

Feature CON-495 - Add new events

bartlomiejmarszal committed:
fix

bartlomiejmarszal committed:
change variable name

Co-authored-by: Gabriel Felipe Soares <gabrielfs7@gmail.com>

bartlomiejmarszal committed:
Fix error on searchAdvanced

bartlomiejmarszal committed:
Merge branch 'fix/CON-496/fix-registering-service' of https://github.com/oat-sa/tao-core into fix/CON-496/fix-registering-service

bartlomiejmarszal committed:
Merge pull request #2791 from oat-sa/fix/CON-496/fix-registering-service

Fix registration process

gabrielfs7 committed:
feat: proxy to search class metadata

Commit:
chore: merge develop

tereshhhchenko committed:
fix: clear search store after login

ADF-1

voknil committed:
Merge develop into feature/TAO-10203-advanced-search

voknil committed:
fix: add proper error handling for search result

boajer committed:
Merge pull request #2820 from oat-sa/feature/ADF-151/fix-compatibility-w-advanced-search

Feature/ADF-151/Added proper error handling for search result
